app/src/org/commcare/dalvik/activities/LoginMode.java

```diff
@@ -0,0 +1,23 @@
+package org.commcare.dalvik.activities;
+
+/**
+ * Enum representing the different logical/UI states that you can be in while on the login screen
+ *
+ * @author Aliza Stone (astone@dimagi.com)
+ */
+public enum LoginMode {
+
+    PASSWORD, PIN, PRIMED;
+
+    public static LoginMode fromString(String s) {
```

Comment:
Isn't needed if you store/restore `LoginMode` using extra serialization: http://stackoverflow.com/a/5309967